Sources/AsyncAlgorithms/AsyncWithLatestFrom2Sequence.swift

```diff
@@ -0,0 +1,261 @@
+              state.other2State = .element(.success(element))
+            }
+            onOther2Element?(element)
+          }
+        } catch {
+          othersState.withCriticalRegion { state in
+            state.other2State = .element(.failure(error))
+          }
+        }
+      }
+    }
+
+    mutating func startOthers() {
+      self.startOther1()
+      self.startOther2()
+    }
+
+    public mutating func next() async rethrows -> Element? {
+      let shouldReturnNil = self.isTerminated.withCriticalRegion { $0 }
+      guard !shouldReturnNil else { return nil }
+
+      self.startOthers()
+
+      return try await withTaskCancellationHandler { [isTerminated, other1Task, other2Task] in
+        isTerminated.withCriticalRegion { isTerminated in
+          isTerminated = true
+        }
+        other1Task?.cancel()
+        other2Task?.cancel()
+      } operation: { [other1Task, other2Task, othersState, onBaseElement] in
+        do {
+          while true {
+            guard let baseElement = try await self.base.next() else {
+              isTerminated.withCriticalRegion { isTerminated in
+                isTerminated = true
+              }
+              other1Task?.cancel()
+              other2Task?.cancel()
+              return nil
+            }
+
+            onBaseElement?(baseElement)
+
+            let decision = othersState.withCriticalRegion { state -> BaseDecision in
+              switch (state.other1State, state.other2State) {
+                case (.notStarted, _):
+                  return .pass
+                case (_, .notStarted):
+                  return .pass
+                case (.idle, _):
+                  return .pass
+                case (_, .idle):
+                  return .pass
+                case (.element(.success(let other1Element)), .element(.success(let other2Element))):
+                  return .returnElement(.success((baseElement, other1Element, other2Element)))
+                case (.element(.failure(let otherError)), _):
+                  return .returnElement(.failure(otherError))
+                case (_, .element(.failure(let otherError))):
+                  return .returnElement(.failure(otherError))
+              }
+            }
+
+            switch decision {
+              case .pass:
+                continue
+              case .returnElement(let result):
+                return try result._rethrowGet()
+            }
+          }
+        } catch {
+          isTerminated.withCriticalRegion { isTerminated in
+            isTerminated = true
+          }
+          other1Task?.cancel()
+          other2Task?.cancel()
+          throw error
+        }
+      }
+    }
+  }
+}
+
+extension AsyncWithLatestFrom2Sequence: Sendable where Base: Sendable, Other1: Sendable, Other2: Sendable { }
+extension AsyncWithLatestFrom2Sequence.Iterator: Sendable where Base.AsyncIterator: Sendable, Other1: Sendable, Other2: Sendable { }
```

Comment:
The `Iterator` really shouldn't be `Sendable` here. The implementation also doesn't need it as far as I can see.